src/Smearing.jl

```diff
@@ -3,12 +3,18 @@
 # Ref for the equations:
 #    - M. Methfessel, A. T. Paxton 1989
 #       "High-precision sampling for Brillouin-zone integration in metals"
+#     - http://www.hector.ac.uk/cse/distributedcse/reports/conquest/conquest/node6.html
+#       "Methfessel-Paxton Approximation to Step Function"
+#     - https://gitlab.com/QEF/q-e/-/blob/bcbb9787c58f17a467653707917172ffc4053dd9/Modules/wgauss.f90
+#       Smearing function implementation in Quantum-Espresso
+#     - https://gitlab.com/QEF/q-e/-/blob/bcbb9787c58f17a467653707917172ffc4053dd9/Modules/w0gauss.f90
+#       Occupation derivative and entropy implementations in Quantum-Espresso
```

Comment:
We tend to also put original references, if we have them handy. Could you add the Cold Smearing paper?